corrected to read from text file

heidikrisse opened the pull request:
The initial code reads a list of integers from a user, but I changed the code to read integers from a text file called 'numbers.txt'. The user still decides the number of the integers. The text file 'numbers.txt' is added.

heidikrisse committed:
corrected to read from text file

heidikrisse committed:
changed to list initialization

learncpp.com advices to use more modern direct list initialization using braces

heidikrisse committed:
changed \n over endl

learncpp.com advises to prefer \n over endl when outputting text to the console

heidikrisse committed:
corrected

Added size type for the vector and removed unnecessary variables average and difference

heidikrisse committed:
functions added

heidikrisse committed:
simplified

heidikrisse committed:
printing into an own function

heidikrisse committed:
using prefix std instead of using directive

soSkary commented on prestudy_assignment.cpp:
```diff
@@ -1,21 +1,26 @@
 #include <algorithm>
+#include <fstream> // for file input/output operations
 
 using namespace std;
```
Avoid "using namespace std;" in global namespace 

https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice

heidikrisse replied:
Thanks! This mistake I also had already noticed and corrected here to my feat branch: https://github.com/heidikrisse/PreStudyAssignment/pull/1/commits/fd1a5d301ad3a3640e9f9895677c035a09695690 :)

soSkary commented on prestudy_assignment.cpp:
```diff
@@ -1,21 +1,26 @@
 int main() {
+    // variable for the number of integers (from the file)
     int n;
     cout << "Enter the number of integers: ";
```
Maybe defeats the point of having the integers in a separate file? If you want to declare the size of std::vector beforehand, then maybe count the file integers first? :)

heidikrisse replied:
Thanks! My mistake, totally misunderstood the instructions, now corrected here https://github.com/heidikrisse/PreStudyAssignment/commit/3992e42e68f8cf6c9a7b1c6f403905daf62e7c0e

soSkary commented on prestudy_assignment.cpp:
```diff
@@ -1,21 +1,26 @@
+
+    // open the file
+    ifstream file("numbers.txt");
     for (int i = 0; i < n; i++) {
```
This can be done without you manually keeping tabs on the number of integers by: 

`int i {0};
while (file >> numbers[i])
{
    ++i;
if (i == n)
    break;
}`

heidikrisse replied:
Thanks soScary! Corrected this one now also here https://github.com/heidikrisse/PreStudyAssignment/commit/3992e42e68f8cf6c9a7b1c6f403905daf62e7c0e
And by the way, you are very skilled at writing concise one-liner code! I noticed this as well when I read your code on GitHub just now. :)

soSkary commented on prestudy_assignment.cpp:
```diff
@@ -30,6 +35,7 @@ int main() {
         if (numbers[i] > max_value) {
             max_value = numbers[i];
         }
+        // initialize a count for the number of times the value appears in the array
```
Doesn't allow for multiple modes :)

heidikrisse replied:
Thanks soScary! I wouldn´t have even noticed to think about this myself, thanks a lot! Now I corrected this one: https://github.com/heidikrisse/PreStudyAssignment/commit/b67b44eaa99f25a226e758540df262f3d877c7db

soSkary reviewed (commented):
Code looking functional and clean! :)

heidikrisse committed:
the headers sorted correctly

The headers sorted alphabetically within the Standard library headers category

heidikrisse committed:
deleted user input and output

corrected not to ask the user to enter the amount of numbers the user wish to calculate from the text file but calculate all the numbers from the text file

heidikrisse committed:
resolved the problem of multiple modes

heidikrisse committed:
Revert "resolved the problem of multiple modes"

This reverts commit 8ec2536fa6a99cabf1a88936abfa454bc8d4c003.

heidikrisse committed:
corrected if multiple mode values
